flask/templating.py

```diff
@@ -101,7 +101,20 @@ def list_templates(self):
 
 
 def _render(template, context, app):
-    """Renders the template and fires the signal"""
+    """Renders the template and fires signals.
+
+    It is possible to override render template in the before_render_template signal.
+    It can be done only if exactly one receiver and it return not None result."""
+
+    brt_resp = before_render_template.send(app, template=template, context=context)
+    
+    overrides = [rv for _, rv in brt_resp if rv is not None]
```

Comment:
Is there some way to intercept function call with decorator? If there is then it's possible to override render functions.
I tried to find some "depth" parameter in the blinker lib but unfortunately it doesn't exist. I think it would not bad feature.